### DISCUSSION TITLE
Issue with Preview not Loading for Images Stored in JSON Column

### DISCUSSION THREAD

**Discussion by paramwork** [filamentphp/filament#9888] [General]
Hello,

I'm encountering an issue with Filament Forms when using FileUpload to handle image uploads. The images are being successfully stored and their file locations are saved correctly, but the previews are failing to load when the image location is stored in a JSON column. However, when I switch to a text-based column for the image location, the preview works correctly.

Here's how my model's `$fillable` property is structured:

```php
protected $fillable = [
    'name',
    'tagline',
    'description',
    'media->logo', // This is the JSON column in question
];
```

The corresponding Filament Form component configuration is as follows:

```php
Forms\Components\FileUpload::make('media.logo')
    ->label('Logo')
    ->image()
    ->disk('b2')
    ->directory('assets/images')
    ->visibility('public')
    ->preserveFilenames(),
```

I've come to understand that there might be limited or no support for JSON columns when it comes to previews in Filament. Could you provide some guidance or a possible workaround for this? 

Using a text-based column does seem to resolve the preview issue, but I'd prefer to keep the JSON column structure if possible.

Any help on this would be greatly appreciated. Thank you!

**Comment by leandrocfe** [comment]
Maybe this?
```php
protected $fillable = [
...
    'media'
];

protected $casts = [
    'media' => 'array'
];
```

**Reply by paramwork** [reply]
It worked. Thanks a lot.
